### PR TITLE
[minor] Fix regex bug

### PR DESCRIPTION
**Pull request by kyfujisa**
# Description

It is not expected to support regular expression in Athenz policy. Should only supports wildcard symbols `*` & `?` during authorization.

## Type of change

- [x] Bug fix
- [x] New feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Non-code changes (update documentation, pipeline, etc.)

### Flags

- [x] breaks backward compatibility
  - I changed the interpretation of the policy, so this is a disruptive change for some use cases.
- [ ] requires a documentation update
- [ ] has untestable code

## Related issue/PR
N/A

---

## Checklist

- [x] Followed the guidelines in the CONTRIBUTING document
- [x] Added prefix `[major]`/`[minor]`/`[patch]`/`[skip]` in the PR title
- [x] Tested and linted the code
- [x] Commented the code
- [x] Made corresponding changes to the documentation
- [x] Confirmed no dropping in test coverage (by [Codecov](https://codecov.io/gh/yahoojapan/athenz-authorizer/pulls))
- [x] Passed all pipeline checking
- [x] Approved by >1 reviewer

## Checklist for maintainer
- [ ] Use `Squash and merge`
- [ ] Double-confirm the merge message has prefix `[major]`/`[minor]`/`[patch]`/`[skip]`
- [ ] Delete the branch after merge

### CODE REVIEW

**Comment by kyfujisa** [comment]
I implemented the logic in [this code](https://github.com/yahoo/athenz/blob/345779c1c9a03f6b70e0102f9e94348b017bf6c9/clients/java/zpe/src/main/java/com/yahoo/athenz/zpe/AuthZpeClient.java#L782-L800) as a reference.

**Comment by WindzCUHK** [comment]
Please also help to update this doc.
https://github.com/yahoojapan/authorization-proxy/blob/master/docs/debug.md

File: policy/assertion_test.go [hunk 1]
```diff
@@ -132,3 +199,108 @@ func TestNewAssertion(t *testing.T) {
...
+			},
+			want: "^\\^\\$\\.\\|\\[\\+\\\\\\(\\)\\{$",
+		},
+		{
+			name: "check to convert wildcards",
+			args: args{
+				glob: ".*wild?card?-*test*",
+			},
+			want: "^\\..*wild.card.-.*test.*$",
+		},
```
Comment (WindzCUHK) [inline policy/assertion_test.go:293]: Please also add test case about `"\*"`. To show escape is not supported.

File: policy/daemon_test.go [hunk 2]
```diff
@@ -938,6 +944,242 @@ func Test_policyd_CheckPolicy(t *testing.T) {
...
+			args: args{
+				ctx:    context.Background(),
+				domain: "dummyDom",
+				roles:  []string{"dummyRole", "dummyRole1"},
+				action: "dummyAct1",
+				// It doesn't match.
+				resource: "dummyRes1234",
+			},
+			want: errors.New("no match: Access denied due to no match to any of the assertions defined in domain policy file"),
+		},
```
Comment (WindzCUHK) [inline policy/daemon_test.go:1182]: Please add a test case about `"\*"`. Escaping wildcard is not supported.

File: policy/assertion_test.go [hunk 3]
```diff
@@ -101,13 +143,38 @@ func TestNewAssertion(t *testing.T) {
 			wantErr: errors.New("assertion format not correct: Access denied due to invalid/empty policy resources"),
 		},
 		{
-			name: "invalid regex",
+			name: "invalid regex, but not return error",
```
Comment (WindzCUHK) [inline policy/assertion_test.go:146]:
```suggestion
			name: "invalid regex, but valid wildcard, not return error",
```

File: policy/assertion_test.go (hunk 1 above)
Comment (kyfujisa) [reply policy/assertion_test.go:293]: Fixed 6c1991f

File: policy/daemon_test.go (hunk 2 above)
Comment (kyfujisa) [reply policy/daemon_test.go:1182]: Fixed bfbaadd

File: policy/assertion_test.go (hunk 3 above)
Comment (kyfujisa) [reply policy/assertion_test.go]: Fixed 15334ae

**Comment by ssunorz** [comment]
Unused variable `replacer`
https://github.com/yahoojapan/athenz-authorizer/blob/15334ae8aa218331d2eb557d55f4032bd15a6722/policy/assertion.go#L27

**Comment by kyfujisa** [comment]
@ssunorz 
thank you, fixed cff259b

**Review by ssunorz — APPROVED** [review]

**Comment by WindzCUHK** [comment]
@kyfujisa
Updated the PR description. Please help to confirm the content.
> It is not expected to support regular expression in Athenz policy. Should only supports wildcard symbols `*` & `?` during authorization.

**Review by WindzCUHK — APPROVED** [review]

**Comment by WindzCUHK** [comment]
LGTM

**Pull request merged**